### PR TITLE
Updates from stacks-blockchain:master

### PR DESCRIPTION
**Pull request by kantai**
This pulls in updates to clarityRef.md and principals.md for correctness.

The files `functions.md` and `install-source.md` were deleted because the content in functions was either pulled into clarityRef, or covered via the generation JSON docs, and the content in install-source is covered in a number of other places.

### CODE REVIEW

File: _core/smart/principals.md [hunk 1]
```diff
@@ -14,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
...
+executes the expression (passed as an argument) with the `tx-sender`
+set to the contract's principal, rather than the current sender. The
+`as-contract` function returns the value of the provided expression.
 
-For example, a smart contract that implements something like a "token faucet" could be implemented as so:
+For example, a smart contract that implements something like a "token
+faucet" could be implemented as so:
 
-```cl
+```scheme
```
Comment (agraebe) [inline _core/smart/principals.md:62]: does scheme work better? I think "cl" is used for most coding sections in the docs

File: _core/smart/principals.md [hunk 2]
```diff
@@ -14,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
...
-```bash
-clarity-cli launch hanomine /data/hano.clar /data/db
+```scheme
+'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR.contract-name
 ```
 
-A smart contract may use the special variable `contract-name` to refer to its own principal.
+For convenience, smart contracts may write a contract's identifier in the
+form `.contract-name`, which is expanded by the Clarity interpreter into
+a fully-qualified contract identifier that corresponds to the same
```
Comment (agraebe) [inline _core/smart/principals.md:50]: This is a mouthful - can we explain in simpler terms and also add why this is relevant? what's the outcome of this behavior?

File: _core/smart/principals.md [hunk 3]
```diff
@@ -14,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
...
+and multi-signature Stacks addresses, a given principal can operate on
+their assets by issuing a signed transaction on the blockchain. _Smart
+contracts_ may also be principals (reprepresented by the smart
+contract's identifier), however, there is no private key associated
+with the smart contract, and it cannot broadcast a signed transaction
+on the blockchain.
+
+A Clarity contract can use a globally defined `tx-sender` variable to
+obtain the current principal. The following example defines a transaction
+type that transfers `amount` uSTX from the sender to a recipient if amount
```
Comment (agraebe) [inline _core/smart/principals.md:28]: what are "uSTX"?

File: _core/smart/principals.md (hunk 1 above)
Comment (kantai) [reply _core/smart/principals.md:62]: neither are perfect, but I've mostly found scheme to work a little better than cl

File: _core/smart/principals.md (hunk 3 above)
Comment (kantai) [reply _core/smart/principals.md:28]: microSTX, will update

Comment (kantai) [reply _core/smart/principals.md]: Updated in 76b9d152

File: _core/smart/principals.md (hunk 2 above)
Comment (kantai) [reply _core/smart/principals.md:50]: I expanded on this a bit in 76b9d152

Comment (agraebe) [reply _core/smart/principals.md:50]: thanks!
